catalyst_voices/lib/pages/registration/registration_dialog.dart

```diff
@@ -56,9 +57,14 @@ class _RegistrationDialogState extends State<RegistrationDialog>
         onPopInvokedWithResult: (didPop, result) {
           unawaited(_confirmedExit(context, didPop: didPop));
         },
-        child: const VoicesTwoPaneDialog(
-          left: RegistrationInfoPanel(),
-          right: RegistrationDetailsPanel(),
+        child: BlocBuilder<RegistrationCubit, RegistrationState>(
+          builder: (context, state) {
```

Comment:
add `buildWhen` so it does not rebuilds every time any of properties changes.

but better it would be better if you just let `VoicesTwoPaneDialog` to accept Widget as parameter which would be used as `closeButton`. Then you could just define `BlocCloseRegistrationButton` with `Offstage` when step `is!  AccountCompletedStep`

Reply:
I've used BlocSelector as Dominik suggested. Lmk if that works for you too.